sh/confirm_new_contract.sh

```diff
@@ -121,14 +121,17 @@ cd "$project_root"
 
 . "$project_root"/sh/common.sh
 
+declare -r target_contract="$1"
+shift
+
 declare safe_address
 safe_address="$(get_config governance.deploymentSafe)"
 declare -r safe_address
 
 . "$project_root"/sh/common_safe.sh
 . "$project_root"/sh/common_safe_owner.sh
 . "$project_root"/sh/common_wallet_type.sh
-. "$project_root"/sh/common_deploy_settler.sh
+. "$project_root"/sh/common_deploy_"$target_contract".sh
```

Comment:
oooooooh.... this is extremely dangerous. definitely validate that `"$target_contract"` is an expected value and consider avoiding parameter expansion in `source` statements altogether

Reply:
now taht you mention it, yes, makes sense that it is indeed dangerous and probably a bad practice. Will think about how to do it in a better way